tests/e2e/performance/test_vdbench_workload.py

```diff
@@ -57,6 +63,20 @@ def teardown():
     worker_nodes = helpers.get_worker_nodes()
     # Getting list of free nodes
     free_nodes = list(set(worker_nodes) - set(ocs_nodes))
+    if not free_nodes:
+        # Getting list of current machinesets
+        log.info('Adding new worker node for the application to run on.')
+        mset = machine.get_machinesets()
```

Comment:
Does this hold good for UPI or VMware configurations? If not, we need a check. I don't see any machineset defined in my vmware cluster?!
```
(.venv) [kramdoss@kramdoss ocs-ci]$ oc get machineset --all-namespaces
No resources found.
(.venv) [kramdoss@kramdoss ocs-ci]$ oc get machinesets -n openshift-machine-api
No resources found.

```

Reply:
This implementation is for AWS IPI.
Yes, i will work on solution for UPI and VMware as well.